exporters/metrics/src/main/java/com.google.cloud.opentelemetry.metric/MetricTranslator.java

```diff
@@ -0,0 +1,143 @@
+import com.google.api.Metric;
+import com.google.api.MetricDescriptor;
+import com.google.cloud.opentelemetry.metric.MetricExporter.MetricWithLabels;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.primitives.Longs;
+import com.google.monitoring.v3.Point;
+import com.google.monitoring.v3.Point.Builder;
+import com.google.monitoring.v3.TimeInterval;
+import com.google.monitoring.v3.TypedValue;
+import com.google.protobuf.Timestamp;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MetricTranslator {
+
+  private static final Logger logger = LoggerFactory.getLogger(MetricTranslator.class);
+
+  static final String DESCRIPTOR_TYPE_URL = "custom.googleapis.com/OpenTelemetry/";
+  static final Set<String> KNOWN_DOMAINS = ImmutableSet
+      .of("googleapis.com", "kubernetes.io", "istio.io", "knative.dev");
+  static final long NANO_PER_SECOND = (long) 1e9;
+
+  static final Set<Type> GAUGE_TYPES = ImmutableSet.of(NON_MONOTONIC_LONG, NON_MONOTONIC_DOUBLE);
+  static final Set<MetricData.Descriptor.Type> CUMULATIVE_TYPES = ImmutableSet
+      .of(MONOTONIC_LONG, MONOTONIC_DOUBLE);
+  static final Set<MetricData.Descriptor.Type> LONG_TYPES = ImmutableSet.of(NON_MONOTONIC_LONG, MONOTONIC_LONG);
+  static final Set<MetricData.Descriptor.Type> DOUBLE_TYPES = ImmutableSet
+      .of(NON_MONOTONIC_DOUBLE, MONOTONIC_DOUBLE);
+
+  static Metric mapMetric(MetricData metric, String type) {
+    Metric.Builder metricBuilder = Metric.newBuilder().setType(type);
+    metric.getDescriptor().getConstantLabels().forEach(metricBuilder::putLabels);
+    return metricBuilder.build();
+  }
+
+  static MetricDescriptor mapMetricDescriptor(MetricData metric) {
+    String instrumentName = metric.getInstrumentationLibraryInfo().getName();
+    MetricDescriptor.Builder builder = MetricDescriptor.newBuilder().setDisplayName(instrumentName)
+        .setType(mapMetricType(instrumentName));
+    metric.getDescriptor().getConstantLabels().forEach((key, value) -> builder.addLabels(mapConstantLabel(key, value)));
+
+    MetricData.Descriptor.Type metricType = metric.getDescriptor().getType();
+    if (GAUGE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.GAUGE);
+    } else if (CUMULATIVE_TYPES.contains(metricType)) {
+      builder.setMetricKind(MetricDescriptor.MetricKind.CUMULATIVE);
+    } else {
+      logger.error("Metric type {} not supported", metricType);
+      return null;
+    }
+    if (LONG_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.INT64);
+    } else if (DOUBLE_TYPES.contains(metricType)) {
+      builder.setValueType(MetricDescriptor.ValueType.DOUBLE);
+    } else {
+      logger.error("Metric type {} not supported", metricType);
```

Comment:
You have the same error message here and above, would be nice to make them more specific (make one mention gauge/cumulative and other long/double?)